ci-operator/step-registry/ibmcloud/provision/vpc/ibmcloud-provision-vpc-commands.sh

```diff
@@ -0,0 +1,196 @@
+    set +o errexit
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    echo "check IBMCLOUD_CLI: ${IBMCLOUD_CLI}..."
+    command -v ${IBMCLOUD_CLI}
+    ${IBMCLOUD_CLI} --version
+    ${IBMCLOUD_CLI} plugin list
+    id
+    echo "Home: $HOME IBMCloud_HOME $IBMCLOUD_HOME"
+    set -o errexit
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    echo "Try to login..."    
+    "${IBMCLOUD_CLI}" login -r ${LEASED_RESOURCE} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function create_resource_group() {
+    local rg="$1"
+    echo "create resource group ... ${rg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg} || return 1
+    "${IBMCLOUD_CLI}" target -g ${rg} || return 1
+}
+
+function getZoneSubnets() {
+    local vpcName="$1" zone="$2"
+
+    "${IBMCLOUD_CLI}" is vpc ${vpcName} --show-attached --output JSON | jq -r --arg z "${zone}" '.subnets[] | select(.zone.name==$z) | .name'
+}
+
+function getZoneAddressprefix() {
+    local vpcName="$1" zone="$2"
+
+    "${IBMCLOUD_CLI}" is vpc-address-prefixes ${vpcName} --output JSON | jq -c -r --arg z ${zone} '.[] | select(.zone.name==$z) | .cidr'
+}
+
+function getOneMoreCidr() {
+    local cidr="$1"
+
+    IFS='.' read -r -a cidr_num_array <<< "${cidr}"
+    cidr_num_array[2]=$((${cidr_num_array[2]} + 1))    
+    IFS=. ; echo "${cidr_num_array[*]}"
+}
+
+function waitAvailable() {
+    local retries=15  try=0 
+    local type="$1" name="$2"
+    while [ "$(${IBMCLOUD_CLI} is ${type} ${name} --output JSON | jq -r '.status')" != "available" ] && [ $try -lt $retries ]; do
+        echo "The vpc is not available, waiting..."
+        sleep 10
+        try=$(expr $try + 1)
+    done
+
+    if [ X"$try" == X"$retries" ]; then
+        echo "Fail to get available ${type} - ${name}"
+        "${IBMCLOUD_CLI}" is ${type} ${name} --output JSON
+        return 1
+    fi    
+}
+
+function create_vpc() {
+    local preName="$1" vpcName="$2" resource_group="$3" region="$4"
+    local zones zone_cidr zone_cidr_main subnetName
+
+    # create vpc
+    "${IBMCLOUD_CLI}" is vpc-create ${vpcName} --resource-group-name ${resource_group}
+
+    waitAvailable "vpc" ${vpcName}
+    
+    # create subnets
+    zones=("${region}-1" "${region}-2" "${region}-3")
+
+    for zone in "${zones[@]}"; do
+        zone_cidr=$(getZoneAddressprefix "${vpcName}" "${zone}")
+        zone_cidr_main="${zone_cidr%/*}"
+        subnetName="${preName}-control-plane-${zone}"
+        "${IBMCLOUD_CLI}" is subnet-create ${subnetName} ${vpcName} --ipv4-cidr-block "${zone_cidr_main}/24"
+        waitAvailable "subnet" ${subnetName}
+    done
+
+    for zone in "${zones[@]}"; do
+        zone_cidr=$(getZoneAddressprefix "${vpcName}" "${zone}")
+        zone_cidr_main=$(getOneMoreCidr "${zone_cidr%/*}")
+        subnetName="${preName}-compute-${zone}"
+        "${IBMCLOUD_CLI}" is subnet-create ${subnetName} ${vpcName} --ipv4-cidr-block "${zone_cidr_main}/24"
+        waitAvailable "subnet" ${subnetName}
```

Comment:
Does this can ensure when subnet creation failed or not ready to use, the function can return a non-zero value to its caller?

Reply:
local checking , when waitAvailable return 1, will exit